pkg/config/resources.go

```diff
@@ -0,0 +1,114 @@
+package config
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"text/template"
+)
+
+// PiperEnvironmentParameter defines a parameter within the Piper environment
+type PiperEnvironmentParameter struct {
+	Name string `json:"name"`
+}
+
+// InfluxResource defines an Influx resouece that holds measurement information for a pipeline run
+type InfluxResource struct {
+	Name         string              `json:"name"`
+	StepName     string              `json:"stepName"`
+	Measurements []InfluxMeasurement `json:"parameters"`
+}
+
+// InfluxMeasurement defines a measurement for Influx reporting which is defined via a step resource
+type InfluxMeasurement struct {
+	Name   string         `json:"name"`
+	Fields []InfluxMetric `json:"fields"`
+	Tags   []InfluxMetric `json:"tags"`
+}
+
+// InfluxMetric defines a metric (column) in an influx measurement
+type InfluxMetric struct {
+	Name string `json:"name"`
+}
+
+// InfluxMetricContent defines the content of an Inflx metric
+type InfluxMetricContent struct {
+	Measurement string
+	ValType     string
+	Name        string
+	Value       *string
+}
+
+// InfluxField is the constant for an Inflx field
+const InfluxField = "field"
+
+// InfluxTag is the constant for an Inflx field
+const InfluxTag = "tag"
+
+const influxStructTemplate = `type {{ .StepName }}{{ .Name | title}} struct {
```

Comment:
I am surprised. Why do we have such a template string here. Are we generating code here which we execute later on directly? Or does this belong to the generator? In this case I would like to suggest to keep that code here somewhere else related to the other generator coding. Having that mixed up would be very confusing.

In case it is not about the generator, I would like to propose to explain some more details ...

Reply:
good point. I moved the stuff to the generator